### ISSUE TITLE
Variable validation for archetype module fails

### ISSUE THREAD

**Issue by larsakerlund** [Azure/terraform-azurerm-caf-enterprise-scale#58]
When creating a custom landing zone hierarchy, validation fails if `_` is present in the map key name. Underscores are supported in a management group ID in ARM, so should be supported by this module as well.

Example hierarchy that doesn't work:
```terraform
custom_landing_zones = {
  (var.root_id) = {
      display_name               = var.root_name
      parent_management_group_id = data.azurerm_client_config.current.tenant_id
      subscription_ids           = []
      archetype_config = {
      archetype_id   = "es_root"
      parameters     = {}
      access_control = {}
      }
  }

  "${var.root_id}_landing_zones" = {
      display_name               = "Landing Zones"
      parent_management_group_id = var.root_id
      subscription_ids           = []
      archetype_config = {
      archetype_id   = "es_landing_zones"
      parameters     = {}
      access_control = {}
      }
  }
}
```

Terraform plan output:
```bash
$ terraform plan

Error: Invalid value for variable

  on .terraform/modules/enterprise_scale/main.tf line 12, in module "management_group_archetypes":
  12:   scope_id                = each.key

The scope_id value must be a valid Subscription or Management Group ID.

This was checked by the validation rule at
.terraform/modules/enterprise_scale/modules/archetypes/variables.tf:21,3-13.
```

Regex validation:
![Screenshot 2021-03-22 at 10 52 25](https://user-images.githubusercontent.com/42139799/111971594-cc5d7280-8afc-11eb-9804-305af7bd0939.png)

**Comment by krowlandson** [comment]
Thank you for your issue @larsakerlund... Following up from my comment on the PR, this limitation was by design as the module is enforcing a more opinionated view on naming standards. We will validate the impact of this on the broader aspects of what this module delivers today, but also what we have planned for the future. We need to ensure this change doesn't impact other resource types which will be managed by future releases of the module to deliver a more complete Enterprise-scale solution.

**Comment by larsakerlund** [comment]
Thank you for the reply @krowlandson, that's great to hear! As you say, this change might cause issues for other types of resources (like subscriptions) if they are also to be considered as a LZ/archetype scope instead of just MGs.

Another thing worth mentioning here is that I noticed when troubleshooting this that the `custom_landing_zone` variable validation only checks the first object key name. This is what made my example above slip through that layer. I suspect that this also might be intentional though, due to limitations with Terraforms input validation.

**Comment by krowlandson** [comment]
@larsakerlund... Just to check, do you consider it important to support underscores in Management Group names, or does your other issue #61 address your primary concern?

Based on the CAF guidance for [Define your naming convention](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-naming) we were excluding underscores as underscores are generally considered as a "word joiner", whilst hyphens represent a "word seperator".

Supporting characters which may cause downstream issues on other resource names is only going to increase the complexity of the code and increase the risk of new issues being introduced.

There's also the consideration that we use hypens in other naming standards throughout the module which would lead to an unsightly mix of both character types in some resource names. Whilst this could be fixed by allowing a `var` to select `hyphen_or_underscore_delimiter`, this is also going to further complicate the code base for something which is emotive rather than technically necessary.

**Comment by larsakerlund** [comment]
This is definitely a minor concern, and not something that has to be fixed at all. It was just my assumption that valid Management Group names was supported by the module. But now knowing your reasoning behind why it is not, I agree with you that it is better to keep this as is. It is not worth the risk.

The fix for #61 would've saved me from troubleshooting this further since it addresses the error.

**Comment by krowlandson** [comment]
That's great, thank you for confirming @larsakerlund. I will close this issue and corresponding PR #59 

We are just running the tests for PR #61 and will have this merged this morning for inclusion in the next release.

Thank you again for your contribution.
